src/validate.js

```diff
@@ -3100,89 +3103,171 @@ function F_VALIDATE(disabledHL) {
-			&& isLimitOk(250)
-			&& address.isOkFor(250))
-			venue.report(250);
+			// check if the venue was already seen
+			if (seen) {
+				HLObject(rawVenue);
+				continue;
+			}
 
-	} // for all venues
+			///////////////////////////////////////////////////////////////////
+			// Prepare simple objects
+			var venue = new SimpleOBJECT(venueID, WMo.venues);
+			Object.seal(venue);
+
+			// mark venue as seen
+			_RT.$seen[venueID] = seen = [0, null, false, false,
+				4 > currentZoom,
+				cityID];
+
+			// increase city counter
+			venue.incCityCounter();
+
+			// shortcuts
+			var address = venue.$address;
+			var country = address.$country;
+			var countryLen = country.length;
+			var countryCode = country ? _I18n.getCountryCode(country.toUpperCase())
+				: _RT.$cachedTopCCode;
+			var city = address.$city;
+			var cityLen = city.length;
+			var cityID = address.$cityID;
+			var street = address.$street;
+			var state = address.$state;
+			var streetLen = street.length;
+			var alts = venue.$alts;
+			var lock = venue.$lock;
+
+			var exceptedCategories ='BRIDGE|ISLAND|FOREST_GROVE|SEA_LAKE_POOL|RIVER_STREAM|CANAL|DAM|TUNNEL|JUNCTION_INTERCHANGE'.split('|');
+
+			if (!cityLen
+				&& exceptedCategories.indexOf(venue.$categories[0]) === -1
+				&& isLimitOk(250)
+				&& address.isOkFor(250))
+				venue.report(250);
+
+			// GROUP name.length
+			if (!venue.$name.length) {
+				if (venue.$categories[0] === "PARK" && !cityLen
+					&& isLimitOk(258)
+					&& address.isOkFor(258))
+					venue.report(258);
+			}
+			// GROUP name.length
+			// Check for last update by bots
+			var botNamesAndIDs = [
+				'^waze-maint', '^105774162$',
+				'^waze3rdparty$', '^361008095$',
+				'^WazeParking1$', '^338475699$',
+				'^admin$', '^-1$',
+				'^avsus$', '^107668852$'
+			];
+			var re = new RegExp(botNamesAndIDs.join('|'),'i');
+
+			if ((re.test(venue.$updatedByID.toString()) || re.test(venue.$updatedBy))
+				&& isLimitOk(251)
+				&& address.isOkFor(251))
+				venue.report(251);
+
+			// GROUP isParkingLot
+			if (venue.$rawObject.isParkingLot()){
+				var catAttr = venue.$categoryAttributes;
+				var parkAttr = catAttr ? catAttr.PARKING_LOT : undefined;
+				// missing parking lot type
+				if ((!parkAttr || !parkAttr.parkingType)
+					&& isLimitOk(252)
+					&& address.isOkFor(252))
+					venue.report(252);
+				// missing cost type
+				if ((!parkAttr || !parkAttr.costType || parkAttr.costType === 'UNKNOWN')
+					&& isLimitOk(253)
+					&& address.isOkFor(253))
+					venue.report(253);
+				// missing payment types
+				if ((parkAttr && parkAttr.costType && parkAttr.costType !== 'FREE'
+					&& parkAttr.costType !== 'UNKNOWN'
+					&& (!parkAttr.paymentType || !parkAttr.paymentType.length))
+					&& isLimitOk(254)
+					&& address.isOkFor(254))
+					venue.report(254);
+				//check elevation of the parking lot
+				if ((!parkAttr || !parkAttr.lotType || parkAttr.lotType.length === 0)
```

Comment:
The checks `!parkAttr.lotType` and `parkAttr.lotType.length === 0` are the same

Reply:
lotType can be undefined or `{}` is that both caught by either check? I thought I needed both checks?

Reply:
> Any value that is not false, undefined, null, 0, NaN, or the empty string (""), and any object, including a Boolean object whose value is false, is considered truthy when used as the condition.

Source: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/if...else

Reply:
If `lotType` is an object, I guess there will be no `length` property there, unless it is added by WME. Do you happen to have a permalink with a parking lot at hand, since there are no parking lots around...

Reply:
Thanks

Reply:
Sure here; [parking lot with things set corectly](https://www.waze.com/nl/editor?env=row&lon=5.92995&lat=52.95845&zoom=5&venues=3867154.38868144.4704730)
and [one with less things set](https://www.waze.com/nl/editor?env=row&lon=5.92995&lat=52.95845&zoom=5&venues=3867154.38933680.13509569)

Reply:
In the correctly set permalink I see the `lotType` is an array and it is empty. Sorry, we need to check if the array length is zero, but I am not quite sure why it is empty in the correctly set PL?

Reply:
Empty? 
![image](https://user-images.githubusercontent.com/63843/46661166-ecad4700-cbb8-11e8-976c-231eaf9c69ae.png)
I see it with one element in it ("STREET_LEVEL")

Reply:
Sorry, I might have missed the venue. We need to check the length of the array `[]` is not zero just as you had it before.